packages/hardhat-core/src/internal/core/jsonrpc/types/input/callRequest.ts

```diff
@@ -21,3 +28,29 @@ export const rpcCallRequest = t.type(
 );
 
 export type RpcCallRequest = t.TypeOf<typeof rpcCallRequest>;
+
+// Types used by eth_call to configure the state override set
+export const stateProperties = t.record(rpcStorageSlotHexStr, rpcStorageSlot);
+
+export const stateOverrideOptions = t.type(
+  {
+    balance: optionalOrNullable(rpcQuantity),
+    nonce: optionalOrNullable(rpcQuantity),
+    code: optionalOrNullable(rpcData),
+    state: optionalOrNullable(stateProperties),
+    stateDiff: optionalOrNullable(stateProperties),
+  },
+  "stateOverrideOptions"
+);
+
+export const stateOverrideSet = t.record(address, stateOverrideOptions);
+export const optionalStateOverrideSet = t.union([
```

Comment:
You can use the `optional` helper, from the same module that `optionalOrNullable` is imported.